genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/JobDirectoryManifest.java

```diff
@@ -262,6 +263,10 @@ private ManifestEntry buildEntry(
             final boolean directory
         ) throws IOException {
             final String path = this.root.relativize(entry).toString();
+            final Path fileName = entry.getFileName();
+            final String name = fileName == null
```

Comment:
Should this error out instead of silently adding an entry with an empty name?

Reply:
It was really put in to avoid the extreme edge case of the `Path` returning `null` and then calling `toString()` on it. The reason I went with empty string is cause that matches the behavior of the `path` attribute above where if you relativize the root to itself it returns empty string. The file name was really only used in this case for display in the UI so it's not critical.